src/transformers/utils/hub.py

```diff
@@ -244,6 +244,9 @@ def try_to_load_from_cache(cache_dir, repo_id, filename, revision=None, commit_h
             with open(os.path.join(model_cache, "refs", revision)) as f:
                 commit_hash = f.read()
 
+    if os.path.isfile(os.path.join(model_cache, ".no_exist", commit_hash, filename)):
+        return -1
```

Comment:
Not necessarily the cleanest here, but I need a return type that is not `None` (means file not found in cache) and not a string (I could put `"no_exist"` but I'm sure someone will end up naming a cached file like this just to spite me).

Reply:
No problem with this as long as there's a comment explaining why!

Reply:
> I could put "no_exist" but I'm sure someone will end up naming a cached file like this just to spite me

For sure :grin: 

For such a case, `-1` is fine but for a more explicit return you can also create an empty object `NO_EXIST` and use it as a return value:

```py
# src/transformers/utils/hub.py

# Return value when trying to load a file from cache but the file does not exist.
NO_EXIST = object() # or "_NO_EXIST"

(...)

def try_to_load_from_cache(cache_dir, repo_id, filename, revision=None, commit_hash=None):
    (...)
    if os.path.isfile(os.path.join(model_cache, ".no_exist", commit_hash, filename)):
        return NO_EXIST

(...)

def cached_file(...):
    (...)
            if resolved_file is not NO_EXIST:
                return resolved_file    
```

Reply:
In any case I agree with @LysandreJik to document it, especially the difference between "a file not existing in the cache (e.g. not cached)" and "a file not existing at all".

Reply:
Just reviewed it and it looks good to me :+1: